src/openrct2/localisation/UTF8.cpp

```diff
@@ -92,6 +92,21 @@ int32_t UTF8Length(const utf8* text)
     return count;
 }
 
+/**
+ * Gets the number of bytes required to store the first x characters / codepoints in a UTF-8 string
+ */
+size_t UTF8Limit(const utf8* text, size_t limit)
+{
+    const utf8* ch = text;
+    const utf8* next_ch = text;
+    while (UTF8GetNext(ch, &next_ch) != 0 && limit > 0)
```

Comment:
UTF8GetNext should have a safe version, this is still prone to read past the buffer.